### DISCUSSION TITLE
Can I add custom html elements to Stencil?

### DISCUSSION THREAD

**Discussion by tkaabbc** [clientIO/joint#2546] [Q&A]
### Introduction

Hi Team,

I want to add the custom element to Stencil, but I have no idea how to pass it through to Stencil.
such as:
```html
<div>
  <label>John Doe</label>
  <input />
</div>
```
<img width="199" alt="image" src="https://github.com/clientIO/joint/assets/35325538/9740621b-b1c0-488d-834c-5923e2878333">

If could give me any tips, I'd be grateful.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
The `ui.Stencil` is a set of papers and each of  the `dia.Paper` renders JointJS shapes. The default shapes are rendered using SVG (not HTML).
- To use HTML components inside your shapes use `<foreignObject />`: https://resources.jointjs.com/tutorial/foreign-object
- Or you can define the whole Stencil as HTML: https://codepen.io/jointjs/details/vYVvjpW
